### ISSUE TITLE
Allow Stage Length to be "hours"

### ISSUE THREAD

**Issue by roryaronson** [openfarmcc/OpenFarm#505]

**Comment by roryaronson** [comment]
And minutes! From a UX interview: "Sowing takes minutes, not days."

**Comment by simonv3** [comment]
How does that relate to the "sow within these days of the year" timeline aspect?

**Comment by roryaronson** [comment]
Good question. I think there are two time things here (and for potentially all Stages) that we need to consider:

1) What is the time range in which this stage should be started? (eg: Sowing should be done between March 15 and April 1)
2) How long does the stage take once it is started? (eg: Sowing takes only 30 minutes to complete)

Currently we ask question (2) outright: "About how long does this stage take?". Which in my opinion is rather user unfriendly compared with the interactive timeline. And we're trying to ask question (1) in the form of the timeline.

How might we ask both questions in the Timeline in a clear way?

I can imagine it getting visually confusing if we tried to use the timeline for both the time range of starting the stage, and the length of the stage.

I want to say that the only important range we need is for sowing, but that's not true. Dormancy or Fruiting may start within a range too!

There must be a visual style that can work to illustrate both clearly...

*to google docs! brb*

**Comment by roryaronson** [comment]
This is getting complicated quick, but I think if we simplify a complex system by removing details then a) we're gonna annoy people who want the detailed info (its useless to them without it) and b) we're taking the easy way out of our job here ;p

Finding a way to include all the details while making it easy to comprehend (by selectively hiding details by default and designing a good layout) is gonna be a challenge, but I think we're up to it!

This is just an idea/brainstorm I had. Feedback and other ideas are welcome. This mockup is for displaying the timeline to the reader. I think that by default, we would display only the bars for 'plant lifetime' and the different stages back-to-back as though it was being planted today. This is most easy to comprehend and actionable right now for the reader.

The power user can optionally turn on the 'stage ranges/windows' mode to see the details of when each stage could happen.

Below, are just different scenarios of the plant being started late or early, and growing fast or slow. Informational diagrams like that may fulfill a 'teachable moment' for the user on how the timeline works.

I also imagine the guide reader being able to play with the timeline and show these scenarios via some sliders like 'anticipated growth speed' and 'anticipated starting date'

Just ideas. Maybe its too complex. I find it useful to diverge to all the possibilities first, then converge on the best.

![timeline idea 2](https://cloud.githubusercontent.com/assets/5524043/6397856/94f4d74a-bd9c-11e4-9878-e175fd7281f6.png)

**Comment by roryaronson** [comment]
What I find most interesting about above diagrams is that if the guide creator supplies the lengths of the stages and the window for the first one, then all other windows, the plant's lifetime, and the growing season can be calculated. #mindblown 

So perhaps an easy way for the guide creator to make this data is by building the timeline where they are asked to choose a range/window for the first stage. Then we show them the rest of the timeline based on their answers to "About how long does this stage last", and ask them to adjust any of the stage lengths if they need to.

**Comment by roryaronson** [comment]
We could also skip out on asking the "how long..." questions outright and just ask for a starting window and the lengths at the timeline step (I like this idea). The author can optionally turn on the 'windows/ranges' of the timeline (which would be calucalted on the fly as they adjust stage lengths) to make sure they get it just right, but we make it simple by default and don't display it.

Ok, I'll be quiet now and wait for other's feedback and ideas ;p

**Comment by roryaronson** [comment]
Google drawings ftw! More ideas about how options could be toggled to make it simple at first glance, but with all the details for the power users:

![more timeline ideas](https://cloud.githubusercontent.com/assets/5524043/6401124/c9e48b42-bdb2-11e4-9542-ad06aa786011.png)

**Comment by md5** [comment]
I have to say as an interested third party that this looks fantastic!

**Comment by md5** [comment]
Here, have some cake: :birthday:

**Comment by roryaronson** [comment]
Haha thanks @md5!

So a little update from my Facebook network and a few teammates here on OpenFarm: People think this looks good. I got some minor tweaks I'll make in an updated mockup in a separate issue. Notably: "growth rate" over-complicates things and is just weird and intangible.

From a Slack conversation with Simon: Sounds like stage *Actions* will each have time associated with them too. eg: the Sow Stage may have Actions: "Water for 15 minutes" and "Sow for 10 minutes". This means that this issue is resolved, as we'll only allow *Stages* to have a minimum length of 1 day, which is what is already the case. Woop!

**Comment by emily-plummer** [comment]
Chiming in late but I like it too! Maybe add tooltips to the time bars so people can hover over the different selections to see what they mean? as a beginning gardener, quickly glancing over the time bars section, I'm not sure what all the different options mean/they all kind of sound like they could mean the same thing to me.

Also, if the plant lifetime & life stage section underneath it are only there to represent how long it the plant's life cycle is, I think it is kind of confusing to see those sections in the context of the year timeline. To me, I think it's confusing to see the life cycle lengths + windows next to each other because it seems like conflicting information. ex: diagram life cycle for sow looks like it should be only late February, but window says late January to early April.

**Comment by roryaronson** [comment]
Thanks for the input @emily-plummer! I added another mockup in #510 with tooltip questions marks for teaching the user what the different options are about. I also simplified down to 3 options, instead of 5. There's also tooltips for the actual bars indicating length, start, and end days.

The sow stage (short brown bar) is how long the stage will actually take in days, eg: sowing will take 5 days. The sowing window (long brown bar) indicates when it is possible to be sowing, or, what range of dates does this have to start and finish in to be successful? Each stage MUST reside within its window. Otherwise the info is conflicting. Maybe this will be clearer in the new mockup. But now you're all spoiled with this prior knowledge! Time to find more people for a UI interview ;p
